### ISSUE TITLE
TCH002 false alarm when importing submodule with import..as

### ISSUE THREAD

**Issue by jakkdl** [charliermarsh/ruff#4667]
## not using import..as
```python
from __future__ import annotations

import libcst
import libcst.matchers

my_list: list[libcst.With] = []
dir(libcst.matchers)
```
```bash
$ ruff --select --isolated TCH002 foo.py
```
no errors, working as expected.

## using import..as
```python
from __future__ import annotations

import libcst
import libcst.matchers as m

my_list: list[libcst.With] = []
dir(m)
```

```bash
$ ruff --select --isolated TCH002 foo.py
foo.py:3:8: TCH002 Move third-party import `libcst` into a type-checking block
Found 1 error.
```

## version
```bash
$ ruff --version
ruff 0.0.269
$ python --version
Python 3.11.3
```

`flake8-type-checking` handles both cases, so this is solely a bug in ruff.
Popped up in https://github.com/Zac-HD/flake8-trio/pull/196#discussion_r1200414885

**Comment by dhruvmanila** [comment]
This seems to be similar to the problem faced in https://github.com/charliermarsh/ruff/issues/4655. I'm guessing the submodule import (`import libcst.matchers`) is increasing the usage count of `libcst` import, thus not flagging for your first example. But, in your second example, the aliased import is not increasing the usage.

**Comment by jakkdl** [comment]
Oh interesting, it's related but also the reverse. So maybe there's two bugs here that cancel out in the first example, where I don't care to create a dedicated `TYPE_CHECKING` block with no performance improvement (since importing `libcst.matchers` will also import `libcst`)

**Comment by charliermarsh** [comment]
Guessing this is a bug in the "strictness" check, whereby we determine whether an import has some other runtime-required import of the same module.

**Comment by charliermarsh** [comment]
@dhruvmanila - Do you wanna take a look? I'd start by looking at `is_implicit_import`.

**Comment by dhruvmanila** [comment]
Yeah, sure!

**Comment by jakkdl** [comment]
Thanks for the quick fix! :rocket:
